packages/issuer-rest/lib/api.ts

```diff
@@ -97,37 +99,82 @@ export class RestAPI {
   }
 
   private registerMetadataEndpoint() {
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore
     this.express.get('/metadata', (request, response) => {
       return response.send(this._vcIssuer._issuerMetadata)
     })
   }
   private registerTokenRequestEndpoint() {
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore
     this.express.post('/token', (request, response) => {
       return RestAPI.sendErrorResponse(response, 501, 'Not implemented')
     })
   }
   private registerCredentialRequestEndpoint() {
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore
-    this.express.post('/credential-request', async (request: CredentialRequest, response) => {
-      return response.send(await this._vcIssuer.issueCredentialFromIssueRequest(request as unknown as CredentialRequest))
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    this.express.get('/credential-request', async (request, _response) => {
+      this._vcIssuer.issueCredentialFromIssueRequest(request as unknown as CredentialRequest)
     })
   }
 
   private registerCredentialOfferEndpoint() {
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore
     this.express.get('/credential-offer/:pre_authorized_code', async (request, response) => {
       const preAuthorizedCode = request.params.pre_authorized_code
       const id = uuidv4()
       this.tokenToId.set(preAuthorizedCode, id)
       return response.send(createCredentialOfferDeeplink(preAuthorizedCode, this._vcIssuer._issuerMetadata))
     })
   }
+
+  private pushedAuthorizationEndpoint() {
+    const handleHttpStatus400 = async (req: Request, res: Response, next: NextFunction) => {
+      if (!req.body) {
+        return res.status(400).send({ error: 'invalid_request', error_description: 'Request body must be present' })
+      }
+      const required = ['client_id', 'code_challenge_method', 'code_challenge', 'redirect_uri']
+      const conditional = ['authorization_details', 'scope']
+      const message = validateRequestBody({ required, conditional, body: req.body })
+      if (message) {
+        return res.status(400).json({
+          error: 'invalid_request',
+          error_description: message,
+        })
+      }
+      return next()
+    }
+
+    this.express.post('/par', handleHttpStatus400, (req: Request, res: Response) => {
+      // Fake client for testing, it needs to come from a registered client
```

Comment:
Please make this a fixme and create a ticket for it

Reply:
Fixed!